chore: add all filters tests

sebastian-palma opened the pull request:
After adding some new code to `all.js` and as per @derberg proposed I thought it'd be better to add some tests in order to not break the existing features.

I just incorporated "Ava" as it's the one used in the generator-filters repository, which seems to work well.

The changes made to `isExpandable` (`(typeof obj.type === "function" && obj.type() === "object")`) are to first check if the object has defined a `type` function, otherwise, it throws an error. But it seems that there are some other parts of an AsyncApi object that might not have `type` defined but some of the other functions that are there checked.

derberg commented on filters/all.contain_no_tag.test.js:
```diff
@@ -0,0 +1,60 @@
+const test = require("ava");
+const {containNoTag} = require("./all");
+
+let undef;
```
this should be `const` as you never modify it. It is also in other test files

derberg commented on filters/all.non_parser_extensions.test.js:
```diff
@@ -0,0 +1,39 @@
+});
+
+test("returns a Map with the truthy schema extensions that do not start with x-parser-", t => {
+  t.context.doc.extensions = function() {
```
not super important, but I just noticed it so writing about it, could be arrow function here

sebastian-palma commented:
Hi @derberg thanks for checking that. I just opted for moving the tests to a separate folder, out of imagination I just named it "tests".

I placed the tests in the same filters folder before because we already have a test folder and to avoid confusion.

Now it seems to be working fine, would you mind giving it another check?

derberg commented:
@sebastian-palma it works fine, just please move those tests to `test` directory, it will be fine, the purpose of this dir is to keep tests and now we just had there a spec that we used for manual testing but we will anyway remove it in near future as we want to work across all the templates with [this](https://www.asyncapi.com/blog/event-streaming-an-additional-architectural-style-to-suplement-api-design/) dummy example to not "abuse" the streetlights example anymore.

could you copy over in your PR this file https://github.com/asyncapi/generator/blob/master/.github/workflows/pull-request-testing.yml so we run tests with each PR? you can leave the lint part and add a script like `"lint": "echo \"No linter specified yet\""` as we should add eslint soon too

sebastian-palma commented:
Everything okay @derberg. I added the workflow file and moved the `tests` files to the `test`. I excluded the folders test/output and test/spec since Ada was running them as tests.

Thanks for the help once again ;)

derberg commented:
@all-contributors please add @sebastian-palma for test
